### PR TITLE
[WFCORE-2233] extend tests for ProcessStateListener

### PR DESCRIPTION
**Pull request by simkam**
* tests for more transitions (reload-required, ....)
* test for execution timetout
* test  for listener that raises exception

https://issues.jboss.org/browse/WFCORE-2233

### CODE REVIEW

**Comment by simkam** [comment]
@ehsavoie fyi

File: testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/events/ProcessStateListenerTestCase.java [hunk 1]
```diff
@@ -84,131 +92,244 @@ static StreamExporter createResourceRoot(Package... additionalPackages) throws I
...
         testSupport = DomainTestSupport.create(DomainTestSupport.Configuration.create(ProcessStateListenerTestCase.class.getSimpleName(),
                 "domain-configs/domain-standard.xml", "host-configs/host-master.xml", null));
+        // add module
         initializeModule(testSupport);
     }
 
+    @AfterClass
+    public static void tearDown() throws Exception {
+        testSupport.stop();
+        PathUtil.deleteSilentlyRecursively(runtimeConfigurationStateChangeFile);
```
Comment (ehsavoie) [inline testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/events/ProcessStateListenerTestCase.java]: Shouldn't we delete the data directory instead ?

**Comment by bstansberry** [comment]
@ehsavoie please approve

**Comment by bstansberry** [comment]
@ehsavoie ok now?

**Comment by ehsavoie** [comment]
:+1:

**Pull request merged**